lib/deploy/hosting/prepare.js

```diff
@@ -12,8 +13,11 @@ module.exports = function(context, options, payload) {
   context.hosting.versionId = context.hosting.versionRef.key();
 
   payload.hosting = options.config.get('hosting');
-  if (!fs.existsSync(resolveProjectPath(options.cwd, payload.hosting.public))) {
+  if (!_.has(payload, 'hosting.public')) {
```

Comment:
This used to throw `Error: An unexpected error has occurred.` if you had no `public` field in your `package.json`. I added this logic to throw better errors and make it clearer what went wrong.